### DISCUSSION TITLE
useBalance causes too many rpc request

### DISCUSSION THREAD

**Discussion by berkaycirak** [wevm/wagmi#3800] [Q&A]
Hi, I am building a dapp where I want to listen user's balance. According to that balance, I am doing some validations on the items (for example, when user want to buy an item and if user has not enough balance then I warn). I make a custom hook to listen balance by following migration guide in wagmi doc
Here is my custom hook 

```js
const useHonBalance = (): {
  honBalance: number;
} => {
  const { address } = useAccount();
  const [honBalance, setHonBalance] = useState(0);
  const chainId = useChainId();
  const currentChainData = getChainById(chainId);
  const queryClient = useQueryClient();

  // Check hon Balance
  const { data: blockNumber } = useBlockNumber({ watch: true });
  const { data: balance, queryKey } = useBalance({
    address: address,
    token: currentChainData?.contractAddresses?.hon as `0x${string}`,
  });

  useEffect(() => {
    queryClient.invalidateQueries({ queryKey });
    if (balance) {
      setHonBalance(+formatUnits(balance.value, 18));
    }
  }, [blockNumber, queryClient, queryKey, balance]);

  return { honBalance };
  ```
  
  
![Screenshot 2024-04-05 at 14 57 13](https://github.com/wevm/wagmi/assets/81809211/a07a77df-84dc-4d5f-9c62-b05134d1c353)

  
  When I use that hook inside my items, there are too many rpc request that slow down my app. Can you help

**Comment by tmm** [comment]
You haven't provided a minimal reproduction so there's no way to help you.
